runtime/vm/jfr.cpp

```diff
@@ -1125,19 +1125,11 @@ jfrSamplingThreadProc(void *entryArg)
 }
 
 jboolean
-setJFRRecordingFileName(J9JavaVM *vm, char *fileName)
+setJFRRecordingFileName(J9JavaVM *vm, char *newFileName)
 {
-	if (NULL != vm->jfrState.jfrFileName) {
-		UDATA defaultFileNameLen = strlen(DEFAULT_JFR_FILE_NAME);
-		const char *jfrFileName = vm->jfrState.jfrFileName;
-		if ((defaultFileNameLen != strlen(jfrFileName))
-				|| (0 != strncmp(DEFAULT_JFR_FILE_NAME, jfrFileName, defaultFileNameLen))
-		) {
-			PORT_ACCESS_FROM_JAVAVM(vm);
-			j9mem_free_memory(vm->jfrState.jfrFileName);
-		}
-	}
-	vm->jfrState.jfrFileName = fileName;
+	PORT_ACCESS_FROM_JAVAVM(vm);
+	j9mem_free_memory(vm->jfrState.jfrFileName);
+	vm->jfrState.jfrFileName = newFileName;
 	VM_JFRWriter::closeJFRFile(vm);
```

Comment:
Suggest closing the file should occur first: for debugging it might want to print the name of the file being closed (which is not reasonable if the name has been freed).